USFMToolsSharp/USFMParser.cs

```diff
@@ -32,15 +34,30 @@ public USFMDocument ParseFromString(string input)
                     continue;
                 }
 
+
                 ConvertToMarkerResult result = ConvertToMarker(match.Groups[1].Value, match.Groups[2].Value);
 
 
+                if(result.marker is TRMarker && !isInsideTable)
+                {
+                    output.Insert(new TableBlock());
+                    isInsideTable = true;
+                }
+                if(isInsideTable && !tempTable.TryInsert(result.marker,checkPossibility:true))

```

Comment:
I don't think you need the `checkPossibility` dry-run logic here.  Since `tempTable` isn't added to the output and is cleared afterwards, there's no impact on the output.  It doesn't seem like you're gaining anything from the added complexity.

Reply:
Since markers are nested, shouldn't you be adding these rows, columns, etc. to a `TableBlock` and then adding it to the output?